xlators/features/shard/src/shard.c

```diff
@@ -1064,23 +1064,26 @@ shard_common_resolve_shards(call_frame_t *frame, xlator_t *this,
     else
         gf_uuid_copy(gfid, local->base_gfid);
 
+    /* Build base shard path before appending index of the shard */
+    strcpy(path, "/" GF_SHARD_DIR "/");
+    uuid_utoa_r(gfid, path + sizeof(GF_SHARD_DIR) + 1);
+    int prefix_len = sizeof(GF_SHARD_DIR) + GF_UUID_BUF_SIZE;
+
     while (shard_idx_iter <= resolve_count) {
         i++;
         if (shard_idx_iter == 0) {
             local->inode_list[i] = inode_ref(res_inode);
             shard_idx_iter++;
             continue;
         }
-
-        shard_make_block_abspath(shard_idx_iter, gfid, path, sizeof(path));
-
+        snprintf(path + prefix_len, sizeof(path) - prefix_len, ".%d",
+                 shard_idx_iter);
```

Comment:
Not strictly necessary, but if you create a function to initialize the `path`, it would be cleaner to also create a function to do this.

Reply:
Yep. Making it a macro